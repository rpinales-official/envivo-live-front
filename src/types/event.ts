// @TODO: revisit this after getting access to the API
export type Event = {
  id: string;
  title: string;
  subtitle?: string;
  imageUrl?: string;
  description?: string;
  ticketUrl?: string;
};