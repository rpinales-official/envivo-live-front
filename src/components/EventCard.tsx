import React from 'react';
import {
    Image,
    Pressable,
    StyleSheet,
    Text,
    View
} from 'react-native';
import { Event } from '../types/event';

type EventCardProps = {
    event: Event;
    onPress?: (id: string) => void;
};

export default function EventCard({
    event,
    onPress,
}: EventCardProps) {
    return (
        <Pressable
            onPress={() => onPress?.(event.id)}
            style={({ pressed }) => [styles.row, pressed && styles.rowPressed]}
        >
            <Image
            // @TODO: replace with a working image
                source={event.imageUrl ? { uri: event.imageUrl } : require('../../assets/placeholder.png')}
                style={styles.image}
            />
            <View style={styles.textWrap}>
                <Text numberOfLines={1} style={styles.title}>{event.title}</Text>
                {!!event.subtitle && (
                    <Text numberOfLines={1} style={styles.subtitle}>{event.subtitle}</Text>
                )}
            </View>
        </Pressable>
    );
}

const ITEM_HEIGHT = 80;

const styles = StyleSheet.create({
    row: {
        minHeight: ITEM_HEIGHT,
        paddingHorizontal: 16,
        paddingVertical: 10,
        backgroundColor: '#0f0f0f',
        flexDirection: 'row',
        alignItems: 'center',
    },
    rowPressed: { opacity: 0.9 },
    image: {
        width: ITEM_HEIGHT - 8,
        height: ITEM_HEIGHT - 8,
        borderRadius: 10,
        backgroundColor: '#1a1a1a',
        marginRight: 12,
    },
    textWrap: {
        flex: 1
    },
    title: {
        color: '#fff',
        fontSize: 16,
        fontWeight: '700',
    },
    subtitle: {
        color: '#cfcfcf',
        fontSize: 12,
        marginTop: 2,
    },
});