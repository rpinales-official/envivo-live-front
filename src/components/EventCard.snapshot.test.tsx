import React from 'react';
import renderer, { act } from 'react-test-renderer';
import EventCard from './EventCard';
import { Event } from '../types/event';

jest.mock('../utils/format', () => ({
  buildEventMeta: () => 'Music Hall Â· Thu 28th 9:00 PM',
}));

describe('EventCard snapshot', () => {
  it('renders title and computed meta', () => {
    const event: Event = {
      id: '1',
      title: 'Rock Night',
      venueName: 'Music Hall',
      startDateTime: '2025-08-13T23:00:00Z',
      timeZone: 'America/New_York',
    };

    let tree: ReturnType<typeof renderer.create>;
    act(() => {
      tree = renderer.create(<EventCard event={event} />);
    });

    expect(tree!.toJSON()).toMatchSnapshot();
  });
});