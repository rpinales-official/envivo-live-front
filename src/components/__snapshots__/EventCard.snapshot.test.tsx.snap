// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard snapshot renders title and computed meta 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#0f0f0f",
        "flexDirection": "row",
        "minHeight": 80,
        "paddingHorizontal": 16,
        "paddingVertical": 10,
      },
      false,
    ]
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/placeholder.png",
      }
    }
    style={
      {
        "backgroundColor": "#1a1a1a",
        "borderRadius": 10,
        "height": 72,
        "marginRight": 12,
        "width": 72,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "700",
        }
      }
    >
      Rock Night
    </Text>
    <Text
      numberOfLines={1}
      style={
        {
          "color": "#cfcfcf",
          "fontSize": 12,
          "marginTop": 2,
        }
      }
    >
      Music Hall Â· Thu 28th 9:00 PM
    </Text>
  </View>
</View>
`;
